src/sqlparser/src/parser.rs

```diff
@@ -268,47 +277,64 @@ impl Parser {
         self.parse_expr().map(WildcardOrExpr::Expr)
     }
 
-    /// Will return a `WildcardExpr::QualifiedWildcard(ObjectName)` with word concat or
-    /// `WildcardExpr::Expr`
+    /// Concats `ident` and `wildcard_expr` in `ident.wildcard_expr`
     pub fn word_concat_wildcard_expr(
         &mut self,
         ident: Ident,
-        expr: WildcardOrExpr,
+        simple_wildcard_expr: WildcardOrExpr,
     ) -> Result<WildcardOrExpr, ParserError> {
-        if let WildcardOrExpr::QualifiedWildcard(mut idents) = expr {
-            let mut id_parts = vec![ident];
-            id_parts.append(&mut idents.0);
-            Ok(WildcardOrExpr::QualifiedWildcard(ObjectName(id_parts)))
-        } else if let WildcardOrExpr::Wildcard = expr {
-            Ok(WildcardOrExpr::QualifiedWildcard(ObjectName(vec![ident])))
-        } else {
-            Ok(expr)
+        let mut idents = vec![ident];
+        match simple_wildcard_expr {
+            WildcardOrExpr::QualifiedWildcard(ids) => idents.extend(ids.0),
+            WildcardOrExpr::Wildcard => {}
+            WildcardOrExpr::ExprQualifiedWildcard(_, _) => unreachable!(),
+            WildcardOrExpr::Expr(e) => return Ok(WildcardOrExpr::Expr(e)),
         }
+        Ok(WildcardOrExpr::QualifiedWildcard(ObjectName(idents)))
     }
 
-    /// Will return a `WildcardExpr::ExprQualifiedWildcard(Expr,ObjectName)` with expr concat or
-    /// `WildcardExpr::Expr`
+    /// Concats `expr` and `wildcard_expr` in `(expr).wildcard_expr`.
     pub fn expr_concat_wildcard_expr(
         &mut self,
         expr: Expr,
-        wildcard_expr: WildcardOrExpr,
+        simple_wildcard_expr: WildcardOrExpr,
     ) -> Result<WildcardOrExpr, ParserError> {
-        if let WildcardOrExpr::QualifiedWildcard(idents) = wildcard_expr {
-            let mut id_parts = idents.0;
-            if let Expr::FieldIdentifier(expr, mut idents) = expr {
-                idents.append(&mut id_parts);
-                Ok(WildcardOrExpr::ExprQualifiedWildcard(*expr, idents))
-            } else {
-                Ok(WildcardOrExpr::ExprQualifiedWildcard(expr, id_parts))
-            }
-        } else if let WildcardOrExpr::Wildcard = wildcard_expr {
-            Ok(WildcardOrExpr::ExprQualifiedWildcard(expr, vec![]))
-        } else {
-            Ok(wildcard_expr)
+        if let WildcardOrExpr::Expr(e) = simple_wildcard_expr {
+            return Ok(WildcardOrExpr::Expr(e));
+        }
```

Comment:
Actually I didn't fully understand the `WildcardOrExpr::Expr` stuff. But it works.... 😇🤡